### PR TITLE
statistics: fix pseudo estimation for correlated column

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

When there's no column statistics, the estimation for `correlated column = column` will be ignored and set to 1.

### What is changed and how it works?

Add calculation for it.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (test with customer's query)

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8734)

### CODE REVIEW

File: cmd/explaintest/r/explain_easy.result [hunk 1]
```diff
@@ -121,11 +121,11 @@ Projection_12	10000.00	root	eq(test.t1.c2, test.t2.c2)
...
   │ └─TableScan_15	10000.00	cop	table:t1, range:[-inf,+inf], keep order:false, stats:pseudo
-  └─Limit_21	1.00	root	offset:0, count:1
-    └─IndexLookUp_43	1.00	root	
-      ├─Limit_42	1.00	cop	offset:0, count:1
-      │ └─IndexScan_40	1.25	cop	table:t2, index:c1, range: decided by [eq(test.t1.c1, test.t2.c1)], keep order:true, stats:pseudo
-      └─TableScan_41	1.00	cop	table:t2, keep order:false
+  └─Limit_21	0.00	root	offset:0, count:1
+    └─IndexLookUp_43	0.00	root	
+      ├─Limit_42	0.00	cop	offset:0, count:1
+      │ └─IndexScan_40	10.00	cop	table:t2, index:c1, range: decided by [eq(test.t1.c1, test.t2.c1)], keep order:true, stats:pseudo
```
Comment (eurekaka) [inline cmd/explaintest/r/explain_easy.result]: Why is the estimated row count of `IndexScan_40` increased?

**Comment by winoros** [comment]
changed @zz-jason

**Pull request merged**